### PR TITLE
feat: Handle BASE64 encoded sourcemaps

### PR DESCRIPTION
**Pull request by loewenheim**
#skip-changelog

### CODE REVIEW

File: crates/symbolicator-service/src/services/sourcemap.rs [hunk 1]
```diff
@@ -161,18 +198,28 @@ impl SourceMapService {
...
-        let sourcemap_file = self
-            .fetch_artifact(source.clone(), sourcemap_artifact.id.clone())
-            .await
-            .map_err(|_| {
-                CacheError::DownloadError("Could not download sourcemap file".to_string())
-            })?;
+        let sourcemap_file = match sourcemap_url {
+            SourceMapUrl::Data(decoded) => {
+                let mut file = NamedTempFile::new().unwrap();
+                file.write_all(&decoded).unwrap();
```
Comment (Swatinem) [inline crates/symbolicator-service/src/services/sourcemap.rs:204]: In the future we should just make this a `ByteView::from_vec`

File: crates/symbolicator-service/src/services/sourcemap.rs [hunk 2]
```diff
@@ -45,6 +46,42 @@ const SOURCEMAP_CACHE_VERSIONS: CacheVersions = CacheVersions {
...
+        match url_string.strip_prefix(Self::DATA_PREAMBLE) {
+            Some(encoded) => {
+                let decoded = BASE64
+                    .decode(encoded.as_bytes())
+                    .map_err(|_| CacheError::Malformed("Invalid base64 sourcemap".to_string()))?;
+                Ok(Self::Data(decoded))
+            }
+            None => {
+                let url = base.join(url_string).map_err(|_| {
+                    CacheError::DownloadError("Invalid sourcemap url: {s}".to_string())
```
[the remark this answers is not in the document]
Comment (loewenheim) [reply crates/symbolicator-service/src/services/sourcemap.rs:77]: Whoops, yeah, that's what I meant. In this branch we *shouldn't* be in the BASE64 case, but I guess we would end up there if the preamble got messed up or something. Should I just leave the url string out of the error message altogether?

File: crates/symbolicator-service/src/services/sourcemap.rs [hunk 3]
```diff
@@ -259,17 +250,14 @@ fn get_release_file_candidate_urls(url: &Url) -> Vec<String> {
 fn resolve_sourcemap_url(
     abs_path: &Url,
     source_artifact: &SearchArtifactResult,
-    mut source_file: &NamedTempFile,
+    source_file: ByteView<'static>,
```
Comment (Swatinem) [inline crates/symbolicator-service/src/services/sourcemap.rs:253]: might as well make this a `&[u8]`

**Review by Swatinem — APPROVED** [review]

**Review by kamilogorek — APPROVED** [review]
Lovely, works like a charm!

**Pull request merged**